flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SynchronousCheckpointITCase.java

```diff
@@ -151,11 +152,15 @@ protected void performDefaultAction(DefaultActionContext context) throws Excepti
 		}
 
 		@Override
-		public boolean triggerCheckpoint(CheckpointMetaData checkpointMetaData, CheckpointOptions checkpointOptions, boolean advanceToEndOfEventTime) throws Exception {
-			eventQueue.put(Event.PRE_TRIGGER_CHECKPOINT);
-			boolean result = super.triggerCheckpoint(checkpointMetaData, checkpointOptions, advanceToEndOfEventTime);
-			eventQueue.put(Event.POST_TRIGGER_CHECKPOINT);
-			return result;
+		public Future<Boolean> triggerCheckpointAsync(CheckpointMetaData checkpointMetaData, CheckpointOptions checkpointOptions, boolean advanceToEndOfEventTime) {
+			try {
+				eventQueue.put(Event.PRE_TRIGGER_CHECKPOINT);
+				Future<Boolean> result = super.triggerCheckpointAsync(checkpointMetaData, checkpointOptions, advanceToEndOfEventTime);
+				eventQueue.put(Event.POST_TRIGGER_CHECKPOINT);
+				return result;
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
```

Comment:
Swallowing of thread interruption. 
Would probably be better to either allow triggerCheckpointAsync to throw (interrupt)exceptions or at least also call Thread.interrupt().

Reply:
This `InterruptedException` is from `LinkiedBlockingQueue.put()` (`eventQueue.put()`).
The queue is not bounded and I hope this exception won't be thrown here during tests runs.
But even if thrown, I'd prefer to propagate them.

Reply:
Yes, do both, interrupt() and throw the exception.
Didn't find the article that I was looking for, but I think this [one](https://www.yegor256.com/2015/10/20/interrupted-exception.html) is good enough.